include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -157,6 +169,33 @@ class LAPACKFullMatrix : public TransposeTable<number>
    */
   void reinit (const size_type size);
 
+  /**
+   * Same as above but will preserve the values of matrix upon resizing.
+   * The original values
+   * of the matrix are kept on increasing the size
+   * \f[
+   * \mathbf A \rightarrow
+   * \left(
+   * \begin{array}{cc}
+   * \mathbf A & \mathbf 0 \\
+   * \mathbf 0 & \mathbf 0
+   * \end{array}
+   * \right)
+   * \f]
+   * Whereas if the new size is smaller, the matrix will contain the upper left block
+   * of the original one
+   * \f[
+   * \mathbf A_{11} \leftarrow
+   * \left(
+   * \begin{array}{cc}
+   * \mathbf A_{11} & \mathbf A_{12} \\
+   * \mathbf A_{21} & \mathbf A_{22}
+   * \end{array}
+   * \right)
+   * \f]
+   */
+  void reinit_preserve (const size_type size);
```

Comment:
Does anyone have a preference regarding the name? I would have preferred `reinit_preserving_entries` or something equally verbose.

Reply:
I am fine with renaming

Reply:
`grow_or_shrink()`?

Reply:
That's a good idea @bangerth, I like `grow_or_shrink` a lot. What do you think @davydden?

Reply:
since both of you like it, i will update 👍

Reply:
done.